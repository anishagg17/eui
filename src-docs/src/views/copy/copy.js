import React, { Component } from 'react';

import {
  EuiCopy,
  EuiButton,
  EuiFieldText,
  EuiSpacer,
  EuiFormRow,
} from '../../../../src/components/';

export default class extends Component {
  constructor(props) {
    super(props);

    this.state = {
      copyText: 'I am the text that will be copied',
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({
      copyText: e.target.value,
    });
  }

  render() {
    return (
      <div>
        <EuiFieldText
          placeholder="Enter text that will be copied to clipboard"
          value={this.state.copyText}
          onChange={this.onChange}
        />

        <EuiSpacer size="m" />

        <EuiCopy textToCopy={this.state.copyText}>
          {copy => (
            <EuiButton onClick={copy}>Click to copy input text</EuiButton>
          )}
        </EuiCopy>
      </div>
    );
  }
}
