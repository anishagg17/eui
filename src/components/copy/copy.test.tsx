import React from 'react';
import { shallow } from 'enzyme';
import { EuiCopy } from './copy';

describe('EuiCopy', () => {
  test('is rendered', () => {
    const component = shallow(
      <EuiCopy textToCopy="some text">
        {copy => <button onClick={copy}>Click to copy input text</button>}
      </EuiCopy>
    );
    expect(component).toMatchSnapshot();
  });

  test('beforeMessage', () => {
    const component = shallow(
      <EuiCopy textToCopy="some text" beforeMessage="copy this">
        {copy => <button onClick={copy}>Click to copy input text</button>}
      </EuiCopy>
    );
    expect(component.state('tooltipText')).toBe('copy this');
  });
});
