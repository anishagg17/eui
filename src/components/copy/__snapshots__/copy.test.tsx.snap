// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiCopy is rendered 1`] = `
<EuiToolTip
  delay="regular"
  onMouseOut={[Function]}
  position="top"
>
  <button
    onClick={[Function]}
  >
    Click to copy input text
  </button>
</EuiToolTip>
`;
